packages/mdc-menu/index.js

```diff
@@ -111,85 +138,106 @@ class MDCMenu extends MDCComponent {
-      deregisterBodyClickHandler: (handler) => document.body.removeEventListener('click', handler),
-      getIndexForEventTarget: (target) => this.items.indexOf(target),
-      notifySelected: (evtData) => this.emit(MDCMenuFoundation.strings.SELECTED_EVENT, {
-        index: evtData.index,
-        item: this.items[evtData.index],
-      }),
-      notifyCancel: () => this.emit(MDCMenuFoundation.strings.CANCEL_EVENT, {}),
-      saveFocus: () => {
-        this.previousFocus_ = document.activeElement;
+      addAttributeToElementAtIndex: (index, attr, value) => {
+        const list = this.items;
+        if (list.length > index) {
+          list[index].setAttribute(attr, value);
+        }
       },
-      restoreFocus: () => {
-        if (this.previousFocus_ && this.previousFocus_.focus) {
-          this.previousFocus_.focus();
+      removeAttributeFromElementAtIndex: (index, attr) => {
+        const list = this.items;
+        if (list.length > index) {
+          list[index].removeAttribute(attr);
         }
       },
-      isFocused: () => document.activeElement === this.root_,
-      focus: () => this.root_.focus(),
-      getFocusedItemIndex: () => this.items.indexOf(document.activeElement),
-      focusItemAtIndex: (index) => this.items[index].focus(),
-      isRtl: () => getComputedStyle(this.root_).getPropertyValue('direction') === 'rtl',
-      setTransformOrigin: (origin) => {
-        this.root_.style[`${getTransformPropertyName(window)}-origin`] = origin;
+      elementContainsClass: (element, className) => element.classList.contains(className),
+      closeSurface: () => this.hide(),
+      getElementIndex: (element) => this.items.indexOf(element),
+      getParentElement: (element) => element.parentElement,
+      getSelectedElementIndex: (selectionGroup) => {
+        return this.items.indexOf(selectionGroup.querySelector(`.${cssClasses.MENU_SELECTED_LIST_ITEM}`));
       },
-      setPosition: (position) => {
-        this.root_.style.left = 'left' in position ? position.left : null;
-        this.root_.style.right = 'right' in position ? position.right : null;
-        this.root_.style.top = 'top' in position ? position.top : null;
-        this.root_.style.bottom = 'bottom' in position ? position.bottom : null;
+      notifySelected: (evtData) => this.emit(strings.SELECTED_EVENT, {
+        index: evtData.index,
+        item: this.items[evtData.index],
+      }),
+      getCheckbox: (index) => {
+        const list = this.items;
+        if (list.length > index) {
+          return list[index].querySelector(strings.CHECKBOX_SELECTOR);
+        } else {
+          return null;
+        }
       },
-      setMaxHeight: (height) => {
-        this.root_.style.maxHeight = height;
+      toggleCheckbox: (checkBox) => {
+        if (checkBox) {
+          checkBox.checked = !checkBox.checked;
+          const event = document.createEvent('Event');
```

Comment:
Why are we emitting a change event here?

This seems like more than what someone would expect of this adapter API... maybe we should put this logic into a util function and call that here and allow wrappers to reuse it, if this is needed?

Reply:
Toggling checked of a checkbox doesn't cause a change event. If someone is listening to the change event of that checkbox to do something with it, it will never get fired if we don't emit the change event.